gateway-js/src/utilities/opentelemetry.ts

```diff
@@ -11,3 +12,16 @@ export enum OpenTelemetrySpanNames {
 
 const { name, version } = require('../../package.json');
 export const tracer = opentelemetry.trace.getTracer(`${name}/${version}`);
+
+export function requestContextSpanAttributes(requestContext: GatewayGraphQLRequestContext): {[key: string]: any} {
+  const spanAttributes: {[key: string]: any} = {};
+
+  if (requestContext.operationName) {
+    spanAttributes["operationName"] = requestContext.operationName;
+  }
+  if (requestContext.source) {
+    spanAttributes["graphql.document"] = requestContext.source;
+  }
+
+  return spanAttributes;
+}
```

Comment:
How about something like this? This incorporates your suggestion to add the semantic naming for a graphql operation, which I think is a good suggestion. We can just keep `operationName` around as deprecated and duplicate it for now.

```suggestion
import type { Attributes } from '@opentelemetry/api';

interface SpanAttributes extends Attributes {
  /**
   * @deprecated in favor of `graphql.operation.name`
   */
  operationName: string | undefined;
  'graphql.operation.name': string | undefined;
  'graphql.document': string;
}

export function requestContextSpanAttributes(
  requestContext: GatewayGraphQLRequestContext,
): SpanAttributes {
  return {
    operationName: requestContext.operationName ?? undefined,
    'graphql.operation.name': requestContext.operationName ?? undefined,
    'graphql.document': requestContext.source,
  };
}
```

Reply:
I like this! I'll change it to use this intermediate span attributes type. 👍

The OpenTelemetry span attribute types allow `undefined` as a value, but [the `Attributes` and `AttributeValue` types are also documented](https://github.com/open-telemetry/opentelemetry-js/blob/2b20565d105d4267de5e19a730c2401eacad43a4/api/src/common/Attributes.ts) as follows:

> [...] the own enumerable keys are counted as valid attribute keys [...]
> [...] null or undefined attribute values are invalid and will result in undefined behavior.

So I'm thinking that this implementation should still avoid setting the key at all if its value will be `undefined`, like in the previous implementation, and not find out what this "undefined behaviour" is.